persistence/sql/publish_to_local_queue.go

```diff
@@ -49,6 +49,21 @@ func (p sqlProcessStoreImpl) doPublishToLocalQueueTx(
 		return nil, err
 	}
 
+	// check if the process is running
+
+	procExecRow, err := tx.SelectProcessExecutionForUpdate(ctx, curProcExecRow.ProcessExecutionId)
+	if err != nil {
+		return nil, err
+	}
+
+	if procExecRow.Status != data_models.ProcessExecutionStatusRunning {
+		return &data_models.PublishToLocalQueueResponse{
+			ProcessExecutionId:  curProcExecRow.ProcessExecutionId,
+			HasNewImmediateTask: false,
+			ProcessNotExists:    false,
```

Comment:
Should we return this error to the client?

Reply:
Here, the process exists, but the status is not running. I feel 404 is not accurate. Maybe 405 Method not allowed?

Reply:
Being silent also sounds good to me because it's no harm to even publish the messages for closed executions on the users' standpoint maybe.

Reply:
It’s not a good practice to silent on this. Because it’s like losing the message . I think 405 is fine

Reply:
An example is that a process failed because of bugs. But messages is being sent from a Kafka topic. The normal expectation is to return error to keep on retrying until server can accept the messages